code/DDSCodeTester.cpp

```diff
@@ -3969,372 +3969,518 @@ class CustomizedDataReaderListener : public DataReaderListener
+        writer_ = publisher_->create_datawriter(topic_, writer_qos);
         //!--
     }
 
     {
         //DDS_CHANGE_RESOURCE_LIMITS_QOS_POLICY
-        ResourceLimitsQosPolicy resource_limits;
-        //The ResourceLimitsQosPolicy is default constructed with max_samples = 5000
-        //Change max_samples to 200
-        resource_limits.max_samples = 200;
-        //The ResourceLimitsQosPolicy is default constructed with max_instances = 10
-        //Change max_instances to 20
-        resource_limits.max_instances = 20;
-        //The ResourceLimitsQosPolicy is default constructed with max_samples_per_instance = 400
-        //Change max_samples_per_instance to 100 as it must be lower than max_samples
-        resource_limits.max_samples_per_instance = 100;
-        //The ResourceLimitsQosPolicy is default constructed with allocated_samples = 100
-        //Change allocated_samples to 50
-        resource_limits.allocated_samples = 50;
+        // This example uses a DataWriter, but it can also be applied to: DataWriter, DataReader and Topic entities
+        DataWriterQos writer_qos;
+        // The ResourceLimitsQosPolicy is default constructed with max_samples = 5000
+        // Change max_samples to 200
+        writer_qos.resource_limits().max_samples = 200;
+        // The ResourceLimitsQosPolicy is default constructed with max_instances = 10
+        // Change max_instances to 20
+        writer_qos.resource_limits().max_instances = 20;
+        // The ResourceLimitsQosPolicy is default constructed with max_samples_per_instance = 400
+        // Change max_samples_per_instance to 100 as it must be lower than max_samples
+        writer_qos.resource_limits().max_samples_per_instance = 100;
+        // The ResourceLimitsQosPolicy is default constructed with allocated_samples = 100
+        // Change allocated_samples to 50
+        writer_qos.resource_limits().allocated_samples = 50;
+        // Use modified QoS in the creation of the corresponding entity
+        writer_ = publisher_->create_datawriter(topic_, writer_qos);
         //!--
     }
 
     {
         //DDS_CHANGE_TOPIC_DATA_QOS_POLICY
-        //The TopicDataQosPolicy is default constructed with an empty vector.
-        TopicDataQosPolicy topic_data;
+        // This example only applies to Topic entities
+        TopicQos topic_qos;
+        // The TopicDataQosPolicy is default constructed with an empty vector.
         std::vector<eprosima::fastrtps::rtps::octet> vec;
-        vec = topic_data.data_vec(); // Getter Function
-
-        //Add two new octets to topic data vector
+        // Add two new octets to topic data vector
         eprosima::fastrtps::rtps::octet val = 3;
         vec.push_back(val);
         val = 10;
         vec.push_back(val);
-        topic_data.data_vec(vec); //Setter Function
+        topic_qos.topic_data().data_vec(vec); // Setter Function
+        // Use modified QoS in the creation of the corresponding Topic
```

Comment:
I don't like this comment throughout. I don't see what value it brings.

Reply:
After thinking about it, it's not that bad, since it sets a clear distinction in the code. You may keep it if you like.